src/prefect/deployments/runner.py

```diff
@@ -650,7 +655,10 @@ def __init__(self, name, tag=None, dockerfile="auto", **build_kwargs):
                 f"Only one tag can be provided - both {image_tag!r} and {tag!r} were"
                 " provided as tags."
             )
-        self.name = image_name
+        namespace, repository = split_repository_path(image_name)
+        if not namespace and PREFECT_DEFAULT_DOCKER_BUILD_NAMESPACE:
+            namespace = PREFECT_DEFAULT_DOCKER_BUILD_NAMESPACE.value()
+        self.name = "/".join(filter(None, [namespace, repository]))
```

Comment:
would it be possible to add a clarifying comment here like 
"if the user hasn't provided a namespace in the `image` param of the `.deploy` method, but they have set the `PREFECT_DEFAULT_DOCKER_BUILD_NAMESPACE` setting, use the namespace from that."

Reply:
maybe it's just me who found it confusing for a moment

Reply:
I added some clarifying comments.